include/openthread/platform/ble.h

```diff
@@ -0,0 +1,278 @@
+    uint16_t mLength; ///< Length of the @p mValue.
+    int8_t   mPower;  ///< Transmit/receive power in dBm.
+} otBleRadioPacket;
+
+/*******************************************************************************
+ * @section Bluetooth Low Energy management.
+ ******************************************************************************/
+
+/**
+ * Enable the Bluetooth Low Energy radio.
+ *
+ * @note BLE Device should use the highest ATT_MTU supported that does not
+ * exceed OT_BLE_ATT_MTU_MAX octets.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @retval OT_ERROR_NONE         Successfully enabled.
+ * @retval OT_ERROR_FAILED       The BLE radio could not be enabled.
+ */
+otError otPlatBleEnable(otInstance *aInstance);
+
+/**
+ * Disable the Bluetooth Low Energy radio.
+ *
+ * When disabled, the BLE stack will flush event queues and not generate new
+ * events. The BLE peripheral is turned off or put into a low power sleep
+ * state. Any dynamic memory used by the stack should be released,
+ * but static memory may remain reserved.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @retval OT_ERROR_NONE        Successfully transitioned to disabled.
+ * @retval OT_ERROR_FAILED      The BLE radio could not be disabled.
+ *
+ */
+otError otPlatBleDisable(otInstance *aInstance);
+
+/****************************************************************************
+ * @section Bluetooth Low Energy GAP.
+ ***************************************************************************/
+
+/**
+ * Starts BLE Advertising procedure.
+ *
+ * The BLE device shall use undirected advertising with no filter applied.
+ * A single BLE Advertising packet must be sent on all advertising
+ * channels (37, 38 and 39).
+ *
+ * @note This function shall be used only for BLE Peripheral role.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ * @param[in] aInterval  The interval between subsequent advertising packets
+ *                       in OT_BLE_ADV_INTERVAL_UNIT units.
+ *                       Shall be within OT_BLE_ADV_INTERVAL_MIN and
+ *                       OT_BLE_ADV_INTERVAL_MAX range.
+ *
+ * @retval OT_ERROR_NONE           Advertising procedure has been started.
+ * @retval OT_ERROR_INVALID_STATE  BLE Device is in invalid state.
+ * @retval OT_ERROR_INVALID_ARGS   Invalid interval value has been supplied.
+ *
+ */
+otError otPlatBleGapAdvStart(otInstance *aInstance, uint16_t aInterval);
+
+/**
+ * Stops BLE Advertising procedure.
+ *
+ * @note This function shall be used only for BLE Peripheral role.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @retval OT_ERROR_NONE           Advertising procedure has been stopped.
+ * @retval OT_ERROR_INVALID_STATE  BLE Device is in invalid state.
+ *
+ */
+otError otPlatBleGapAdvStop(otInstance *aInstance);
+
+/**
+ * The BLE driver calls this method to notify OpenThread that a BLE Device has
```

Comment:
Should we say here that a BLE Central device has connected? E.g.

```suggestion
 * The BLE driver calls this method to notify OpenThread that a BLE Central Device has
```

Reply:
fixed.